src/stores/simulation.ts

```diff
@@ -45,6 +45,7 @@ export const simulationDataset = gqlSubscribable<SimulationDataset | null>(
   { simulationDatasetId },
   null,
   null,
+  ({ extent, ...etc }) => ({ ...etc, extent: extent?.extent ?? null }),
```

Comment:
Is this a `SimulationDataset`?
```({ extent, ...etc })```

If it is, then because you're already pulling out `extent`, which is a `string`, the following `extent?.extent` will always be `undefined`, thus the returned SimulationDataset will always have a `null` `extent`.

I suggest adding the type to the argument (i.e. `({extent, ...etc}: SimulationDataset)` if that's accurate).

Also, if it makes sense to you, I noticed that the type of `extent` is already potentially `null`, so you theoretically don't have to do anything other than just return the object.

Reply:
Ok...maybe i should read the whole PR before making comments 🤦🏻 ... I just noticed that it's not technically a `SimulationDataset` yet. I'd suggest making a new type and setting it to that instead of `SimulationDataset` like my previous dumb self suggested...just for extra type security.